platform-sdk/consensus-otter-docker-app/src/testFixtures/java/org/hiero/consensus/otter/docker/app/OutboundDispatcher.java

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: Apache-2.0
+package org.hiero.consensus.otter.docker.app;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import io.grpc.stub.ServerCallStreamObserver;
+import io.grpc.stub.StreamObserver;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.hiero.otter.fixtures.container.proto.EventMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handles queuing {@link EventMessage}s and delivering them to a gRPC {@link StreamObserver} on a
+ * single background thread. This keeps {@code DockerInit} slim and separates concerns between
+ * application initialisation and outbound message handling.
+ */
+public final class OutboundDispatcher {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OutboundDispatcher.class);
+
+    /** Queue used to hand over messages from the platform threads to the dispatcher thread. */
+    private final BlockingQueue<EventMessage> outboundQueue = new LinkedBlockingQueue<>();
+
+    /** Indicates whether the dispatcher has been cancelled. */
+    private final AtomicBoolean cancelled = new AtomicBoolean(false);
+
+    /** Handle to the running dispatcher task so it can be cancelled. */
+    private final Future<?> dispatchFuture;
+
+    /**
+     * Creates a new dispatcher instance and immediately starts the background task.
+     *
+     * @param executor the executor used to run the background task
+     * @param responseObserver the gRPC stream to which messages will be delivered
+     */
+    public OutboundDispatcher(
+            @NonNull final ExecutorService executor, @NonNull final StreamObserver<EventMessage> responseObserver) {
```

Comment:
It looks like an `Executor` instead of an `ExecutorService` would be sufficient. (The former has the advantage that it can be mocked with a lambda expression in tests.)